crypto/fipsmodule/cpucap/cpu_aarch64_apple.c

```diff
@@ -99,11 +99,9 @@ void OPENSSL_cpuid_setup(void) {
     OPENSSL_armcap_P |= ARMV8_APPLE_M1;
   }
 
-#if defined(MAKE_DIT_AVAILABLE)
   if (has_hw_feature("hw.optional.arm.FEAT_DIT")) {
-    OPENSSL_armcap_P |= ARMV8_DIT;
+    OPENSSL_armcap_P |= (ARMV8_DIT | ARMV8_DIT_ALLOWED);
   }
-#endif  // MAKE_DIT_AVAILABLE
 
```

Comment:
Why remove the `MAKE_DIT_AVAILABLE` guard?

Reply:
To enable the CPU capability check regardless of the build flag. 
I had missed to save the same change in `cpu_aarch64_liunx.c` (that's what I forced pushed).
Now, the build flag is only responsible for making the macro at the beginning of the APIs effective.

Reply:
Ok, what the `MAKE_DIT_AVAILABLE` build flag does now is no longer to "make DIT available". Instead it determines whether our internal `SET_DIT_AUTO_DISABLE` macro is a no-op or not.  

It seems like a better name for the flag might be something like `AUTO_ENABLE_DISABLE_SET_DIT`?